src/parser/parser.yy

```diff
@@ -566,112 +569,220 @@ expression
             $$ = UnaryExpression::makeNegate(qctx->objPool(), $3);
         }
         scanner.setUnaryMinus(false);
+        if(!$$->checkDepth()){
+            throw nebula::GraphParser::syntax_error(@1, "The above expression is not a valid expression, because its depth exceeds the maximum depth!");
+        }
```

Comment:
I am sure is it better to wrap the check as a macro function. Writing the same check and error messages too many times seems not well.

Reply:
done